app/bot/openai_test.go

```diff
@@ -0,0 +1,88 @@
+package bot
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/radio-t/super-bot/app/bot/mocks"
+	"github.com/sashabaranov/go-openai"
+	"github.com/stretchr/testify/assert"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestOpenAI_Help(t *testing.T) {
+	require.Contains(t, (&OpenAI{}).Help(), "chat!")
+}
+
+func TestOpenAI_OnMessage(t *testing.T) {
+	// Example of response from OpenAI API
+	// https://platform.openai.com/docs/api-reference/chat
+	mockFunc := func(ctx context.Context, request openai.ChatCompletionRequest) (openai.ChatCompletionResponse, error) {
+		var response openai.ChatCompletionResponse
+
+		jsonResponse := `{
+		  "id": "chatcmpl-123",
+		  "object": "chat.completion",
+		  "created": 1677652288,
+		  "choices": [{
+			"index": 0,
+			"message": {
+			  "role": "assistant",
+			  "content": "Mock response"
+			},
+			"finish_reason": "stop"
+		  }],
+		  "usage": {
+			"prompt_tokens": 9,
+			"completion_tokens": 12,
+			"total_tokens": 21
+		  }
+		}`
+
+		err := json.Unmarshal([]byte(jsonResponse), &response)
+		if err != nil {
+			fmt.Printf("Error: %s", err)
+		}
+		return response, nil
+	}
+
+	mockOpenAIClient := &mocks.OpenAIClient{CreateChatCompletionFunc: mockFunc}
+
+	o := &OpenAI{
+		authToken: "ss-mockToken",
+		client:    mockOpenAIClient,
+	}
+
+	mockMessage := Message{Text: "chat! Mock request to ChatGPT"}
```

Comment:
i don't see why mockMessage should be a separate var here, would put it directly to assert